drivers/usb/device/usb_dc_sam0.c

```diff
@@ -234,7 +247,10 @@ int usb_dc_attach(void)
 	(void)memset(data->descriptors, 0, sizeof(data->descriptors));
 	regs->DESCADD.reg = (uintptr_t)&data->descriptors[0];
 
-	regs->INTENSET.reg = USB_DEVICE_INTENSET_EORST;
+	regs->INTENSET.reg = USB_DEVICE_INTENSET_EORST |
+			     USB_DEVICE_INTENSET_EORSM |
+			     USB_DEVICE_INTENSET_SUSPEND |
+			     USB_DEVICE_INTENSET_WAKEUP;
```

Comment:
Is this interrupt used for Remote Wakeup up as described in the datasheet 32.6.2.14?

Reply:
We probably have different datasheets. TBH I do not know how a device could use this WAKEUP interrupt in opposite to EORSM and SUSPEND. Remote Wakeup feature is to wake-up the host (29.6.3.7 Suspend and Wakeup, SAM R21)

Reply:
Apologies, I should have indicated I was referring to the SAMD21 datasheet.  I think the equivalent section in the SAMR21 datasheet is 29.6.2.14 Remote Wakeup.

My understanding is the device can initiate a remote wakeup of the host?

Reply:
Yes, like HID keyboard can wake-up a PC. We actually have API for that, usb_dc_wakeup_request(), see also usb_dc_nrfx driver.

Reply:
So, it is not clear to me, this is ok then?

Reply:
No

Reply:
So what is the issue?

Reply:
>You do not need to enable WAKEUP in the device mode, please remove it here and above L118..L122

Reply:
Why?  From the samd21 datasheet:

> When the controller sends the Upstream Resume INTFLAG.WAKEUP is set and INTFLAG.SUSPEND is cleared.
The CTRLB.UPRSM is cleared at the end of the transmitting Upstream Resume.

It appears that this indication has a useful purpose?

Reply:
Since I do not use the WAKEUP interrupt notification I will go ahead and remove it.  I put it in there because it appears to have some useful purpose.  So, I do think you should consider that it should be supported in the future.

Reply:
As I already annotated [here](https://github.com/zephyrproject-rtos/zephyr/pull/24402#discussion_r409687415), this is something for the host mode, see "Host Operations". It probably can also be used in the "Upstream Resume" procedure, but is is not the way you wanted to use it. Please remove WAKEUP comment from the commit message.